serverless/template.yml

```diff
@@ -82,60 +94,106 @@ Resources:
                   - "logs:PutLogEvents"
                 Resource:
                   Fn::Sub: "arn:aws:logs:${AWS::Region}:${AWS::AccountId}:log-group:/aws/lambda/PrometheusTimestreamConnector-LambdaFunction-*:*"
+              - Effect: "Allow"
+                Action:
+                  - "timestream:DescribeEndpoints"
+                Resource: "*"
+              - Effect: "Allow"
+                Action:
+                  - "timestream:WriteRecords"
+                  - "timestream:Select"
+                Resource: !Sub "arn:aws:timestream:${AWS::Region}:${AWS::AccountId}:database/${DefaultDatabase}/table/${DefaultTable}"
 
   LambdaFunction:
     Type: "AWS::Serverless::Function"
     Properties:
-      Role: !GetAtt "IAMLambdaRole.Arn"
+      Role: !GetAtt IAMLambdaRole.Arn
       CodeUri: 
-          Bucket: !Sub 'timestreamassets-${AWS::Region}'
-          Key: "timestream-prometheus-connector/timestream-prometheus-connector-linux-amd64-1.2.0.zip"
+        Bucket: !Sub 'timestreamassets-${AWS::Region}'
+        Key: "timestream-prometheus-connector/timestream-prometheus-connector-linux-amd64-1.2.0.zip"
       Description: "Prometheus remote storage connector for Amazon Timestream"
       Handler: "bootstrap"
-      MemorySize: !Ref "MemorySize"
+      MemorySize: !Ref MemorySize
       Timeout: !Ref LambdaTimeoutInSeconds
       Runtime: "provided.al2023"
       Environment:
         Variables:
-          default_database: !Ref "DefaultDatabase"
-          default_table: !Ref "DefaultTable"
-          region: !Ref "AWS::Region"
-          log_level: !Ref "LogLevel"
-      Events:
```

Comment:
`AWS::Serverless::HttpApi` has an `Auth` property that can be used to set SigV4 authentication for the API Gateway. However, it must use a hardcoded `boolean`. Returned values from `!If` or `!Equals` don't work. To get around this, the deployment of the API Gateway has been made more verbose, and the events have been removed from the Lambda function, so that routes can be defined and configured with strings to enable SigV4 authentication (`AWS_IAM` or `NONE`).

Despite all of these changes, and the API Gateway being turned into an `AWS::ApiGatewayV2::Api` resource, this should all behave identically to our previous configuration.